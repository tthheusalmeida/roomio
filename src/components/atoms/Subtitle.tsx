import { mergeClassNames } from "@/utils/classNames";

interface TitleProps {
  children: React.ReactNode;
  className?: string;
}

export default function Subtitle({ children, className }: TitleProps) {
  return (
    <>
      <h2 className={mergeClassNames("font-bold text-2xl my-8", className)}>
        {children}
      </h2>
    </>
  );
}
