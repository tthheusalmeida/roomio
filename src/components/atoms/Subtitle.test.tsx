import { render, screen } from "@testing-library/react";
import Subtitle from "./Subtitle";

describe("Subtitle component", () => {
  it("renders the children text correctly", () => {
    const { asFragment } = render(<Subtitle>My Subtitle</Subtitle>);

    const heading = screen.getByText("My Subtitle");
    expect(heading).toBeInTheDocument();
    expect(heading.tagName).toBe("H2");

    expect(asFragment()).toMatchSnapshot();
  });

  it("applies extra classes passed via className", () => {
    const { asFragment } = render(
      <Subtitle className="extra-class">My Subtitle</Subtitle>
    );

    const heading = screen.getByText("My Subtitle");
    expect(heading).toHaveClass("extra-class");

    expect(asFragment()).toMatchSnapshot();
  });
});
