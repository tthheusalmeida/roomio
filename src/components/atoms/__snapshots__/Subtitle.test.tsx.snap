// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Subtitle component applies extra classes passed via className 1`] = `
<DocumentFragment>
  <h2
    class="font-bold text-2xl my-8 extra-class"
  >
    My Subtitle
  </h2>
</DocumentFragment>
`;

exports[`Subtitle component renders the children text correctly 1`] = `
<DocumentFragment>
  <h2
    class="font-bold text-2xl my-8"
  >
    My Subtitle
  </h2>
</DocumentFragment>
`;
